signer/src/error.rs

```diff
@@ -21,6 +35,11 @@ pub enum Error {
     #[error("tokio i/o error: {0}")]
     TokioIo(#[from] tokio::io::Error),
 
+    /// Error occurred when constructing a BitcoinCoreClient. This takes a String
+    /// as [`sbtc::error::Error`] is not [`Sync`] + [`Send`].
+    #[error("failed to construct a BitcoinCoreClient: {0}")]
+    BitcoinCoreClient(String),
```

Comment:
Is that still the case even after removing the `sbtc::error::Error::BitcoinClient` variant? If not, I wonder if making that error variant take a `Box<dyn std::error::Error + Send + Sync + 'static>` was also a fix. You'd probably have to change the associated error type of the `BitcoinClient`.

Reply:
Now `sbtc::error::Error` is `Sync + Send` -- I don't think I use this specific variant anymore as I added concrete variants, will check. I do think I tried adding `Sync + Send` to the declaration and it still didn't help, as well as using `Arc` instead of `Box`.

Reply:
Hm, I must have ended up in some sort of mixed situation because when I went back and tried it again cleanly it worked. So I've updated this now to retain the error variant :+1: